eZ/Publish/Core/FieldType/Tests/ImageAsset/MapperTest.php

```diff
@@ -0,0 +1,309 @@
+
+        $mapper = $this->createPartialMapper(['isAsset']);
+        $mapper
+            ->expects($this->once())
+            ->method('isAsset')
+            ->with($content)
+            ->willReturn(false);
+
+        $mapper->getAssetField($content);
+    }
+
+    public function testGetAssetFieldDefinition()
+    {
+        $fieldDefinition = new FieldDefinition();
+
+        $contentType = $this->createMock(ContentType::class);
+        $contentType
+            ->expects($this->once())
+            ->method('getFieldDefinition')
+            ->with($this->mappings['content_field_identifier'])
+            ->willReturn($fieldDefinition);
+
+        $this->contentTypeService
+            ->expects($this->once())
+            ->method('loadContentTypeByIdentifier')
+            ->with($this->mappings['content_type_identifier'])
+            ->willReturn($contentType);
+
+        $this->assertEquals($fieldDefinition, $this->createMapper()->getAssetFieldDefinition());
+    }
+
+    public function testGetAssetValue()
+    {
+        $expectedValue = new Image\Value();
+        $content = $this->createContentWithId(self::EXAMPLE_CONTENT_ID);
+
+        $mapper = $this->createPartialMapper(['isAsset']);
+        $mapper
+            ->expects($this->once())
+            ->method('isAsset')
+            ->with($content)
+            ->willReturn(true);
+
+        $content
+            ->expects($this->once())
+            ->method('getFieldValue')
+            ->with($this->mappings['content_field_identifier'])
+            ->willReturn($expectedValue);
+
+        $this->assertEquals($expectedValue, $mapper->getAssetValue($content));
+    }
+
+    /**
+     * @expectedException \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    public function testGetAssetValueThrowsInvalidArgumentException()
+    {
+        $content = $this->createContentWithId(self::EXAMPLE_CONTENT_ID);
+
+        $mapper = $this->createPartialMapper(['isAsset']);
+        $mapper
+            ->expects($this->once())
+            ->method('isAsset')
+            ->with($content)
+            ->willReturn(false);
+
+        $mapper->getAssetField($content);
+    }
+
+    /**
+     * @dataProvider dataProviderForIsAsset
+     */
+    public function testIsAsset(int $contentContentTypeId, int $assetContentTypeId, bool $expected)
+    {
+        $assetContentType = new ContentType([
+            'id' => $assetContentTypeId,
+        ]);
+
+        $this->contentTypeService
+            ->expects($this->once())
+            ->method('loadContentTypeByIdentifier')
+            ->with($this->mappings['content_type_identifier'])
+            ->willReturn($assetContentType);
+
+        $actual = $this
+            ->createMapper()
+            ->isAsset($this->createContentWithContentType($contentContentTypeId));
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    public function dataProviderForIsAsset(): array
```

Comment:
Should we add docblock here and below (`createMapper()`, `createPartialMapper()`, `createContentWithId()`, `createContentWithContentType()`?